net.sonic_pi.SonicPi.json

```diff
@@ -205,16 +336,253 @@
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://github.com/sonic-pi-net/sonic-pi/archive/v3.3.1.tar.gz",
-                    "sha256": "6aab8e22daff1832b7af8221d7d124e4ae0adeaf1c726d8f62a8d3d57bd27d9c"
+                    "url": "https://github.com/sonic-pi-net/sonic-pi/archive/refs/tags/v4.2.0.tar.gz",
+                    "sha256": "22376b5acf528582c6b8f43a6c53df368775ac4cc7b1136fdfc073c05d2ad56d"
                 },
                 {
                     "type": "patch",
-                    "path": "sonic-pi-fixes.patch"
+                    "path": "patches/sonic-pi-build-fixes.patch"
                 },
                 {
-                    "type": "patch",
-                    "path": "sonic-pi-sc-path.patch"
+                    "type": "file",
+                    "url": "https://repo.hex.pm/installs/1.13.0/hex-1.0.1.ez",
+                    "dest-filename": "hex.ez",
+                    "sha256": "84bb9186d87ce91457b2aaaf5bf2d82cbe971240ee8e405228c8bdbb6af699fb"
+                },
+                {
+                    "type": "file",
+                    "url": "https://repo.hex.pm/installs/1.13.0/rebar3-3.15.2",
+                    "dest-filename": "rebar3",
+                    "sha256": "9c9d0fa142e60711147d23de7520fab4a8e32608fad5fddfbcb764e4ce98505f"
+                },
+                /* elixir dependencies */
```

Comment:
if these are generated by a script then it should be in a separate .json. Easier for maintenance.